sklearn/neighbors/__init__.py

```diff
@@ -3,7 +3,7 @@
 algorithm.
 """
 
-from ._ball_tree import BallTree
+from ._ball_tree import BallTree64 as BallTree
```

Comment:
I think this can be done for the purposes of this PR as we will be adding these classes anyways in the follow ups.

Reply:
I agree, ultinately `BallTree` and `KDTree` (as public API) must not inherit from any implementation class.

Reply:
This looks good.

> However the BallTree and KDTree would not be inheriting from the 64 versions
We can change the actual implementations later on without much worry since it'll be opaque to the user (no real UX change until after deprecation)